app/components/home/ProductCard.tsx

```diff
@@ -8,7 +8,7 @@ import Link from 'next/link';
 
   const ProductCard: React.FC<ProductCardProps> = ({ product }) => {
     return (
-      <Link href={`/products/${product.product_id}`} passHref>
+      <Link href={`/product/${encodeURIComponent(JSON.stringify(product))}`} passHref>
```

Comment:
hmmm... it is a good practice to wrap JSON.stringify in a try-catch block, especially if the structure of the object or content is dynamic and might change. This helps you catch and handle any errors gracefully
So, please move this wrap to another place instead to do in the href